source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -147,7 +147,16 @@ void EnvoyQuicClientStream::encodeData(Buffer::Instance& data, bool end_stream)
       // TODO(danzh): investigate the cost of allocating one buffer per slice.
       // If it turns out to be expensive, add a new function to free data in the middle in buffer
       // interface and re-design QuicheMemSliceImpl.
-      quic_slices.emplace_back(quiche::QuicheMemSlice::InPlace(), data, slice.len_);
+      if (!Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.quiche_use_mem_slice_releasor_api")) {
+        quic_slices.emplace_back(quiche::QuicheMemSlice::InPlace(), data, slice.len_);
+      } else {
+        auto single_slice_buffer = std::make_unique<Buffer::OwnedImpl>();
+        single_slice_buffer->move(data, slice.len_);
+        quic_slices.emplace_back(
+            reinterpret_cast<char*>(single_slice_buffer->frontSlice().mem_), slice.len_,
+            [single_slice_buffer = std::move(single_slice_buffer)](const char*) {});
```

Comment:
nit: Can we add a comment here to explain what's going on. At first glance, it looks like we move the data into a new unique_ptr<Buffer>, and then pass a pointer to the data as the first argument to this function and then the local variable goes out of scope and deletes the data. But of course, thats' not what happens because in the lambda capture, we std::move() that unique ptr into a captured variable which lives until that invokable goes out of scope when the slice is destructed. Do I have that right? 

In other words, the only purpose of the callback is to capture the unique_ptr, not actually to *do* anything when executed (The action happens when the callback is destroyed)? We could also explicitly clear `single_slice_buffer` in the body of the callback if we wanted to be a bit more explicit, though I'm not sure how useful that is.

Reply:
I think clearing that explicitly is inherently more readable and makes it easier to reason about the lifetime (since then it's tied to the invocation of the releasor, as opposed to the destructor). That's not a bad idea.